bash_archiving.sh

```diff
@@ -1,117 +1,124 @@
+
+						date=$( date +"_%d_%m_%y-%H_%M_%S" )
+
+						if [[ $(find /$folder/ -type f -user $1 2>/dev/null) ]]
+						then
+							mkdir -p "$1_$folder"
+							sudo find /$folder/ -type f -user $1 -exec cp -rp {} "$1_$folder"/ 2>/dev/null \;
+							tar -czf "$1_$folder$date.tgz" "$1_$folder"/
+							echo -e "\n$folder.tgz contain this files:\n"
+							tar -ztvf "$1_$folder$date.tgz"
+							rm -rf "$1_$folder" 2>/dev/null
+						else
+							continue
+						fi
+					done
+				done < config
+			fi
 		done
-		done 
-   		fi
-	done
-		
-		if [ "${x}" == 0 ] && [ "$1" != "$word" ]; then 
-	        	echo "User $1 don't exist"
-			x=0 
-			exit 2;
-                fi
-
-       
+
+		if [ "${x}" == 0 ] && [ "$1" != "$word" ]
+		then
+			echo "User $1 don't exist"
+			x=0
+			exit 2
+		fi
 	elif (($# == 2 ))
 	then
-			
-if [[ "$2" =~ ([0-9]*minutes|hours|days) ]];then
-			
-if [[ "$2" =~ [0-9]*hours ]];then
-			
-minutes=$( echo $2 | grep -o '[0-9]*' )
-minutes=$(($minutes * 60))
-			
-
-
-elif [[ "$2" =~ [0-9]*days ]];then
- 
-minutes=$( echo $2 | grep -o '[0-9]*' )
-minutes=$(($minutes * 1440)) #minutes in day
-                        
-else 
-			
-minutes=$( echo $2 | grep -o '[0-9]*' )			
-fi
-			
-for (( index=1; index <= $countUSERS; index++))
-do
-word=$(cat /etc/passwd | sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' | tr '\n' ' ' | cut -d " " -f $index) 
-if [ "$1" == "$word" ];then
-x=1 #user founded        
-cat config | while read line; do
-line=$(echo  $line | sed 's/ *//g')
-			
-IFS=:
-for i in $line; do	
-folder=$(echo "$i" | cut -c 2- 2>/dev/null | tr -d '\n' | tr -d '^') 
-			
-date=$( date +"_%d_%m_%y-%H_%M_%S" )
-                
-if [[ $(find /$folder/ -type f -user $1 -amin "-$minutes" 2>/dev/null;) ]];then 	
-mkdir -p "$1_$folder";
-sudo find /$folder/ -type f -user $1 -amin "-$minutes" -exec cp -rp {} "$1_$folder"/ 2>/dev/null \; 
-tar -czf "$1_$folder$date.tgz" "$1_$folder"/ 
-echo -e "\n$folder.tgz contain this files:\n" ;
-tar -ztvf "$1_$folder$date.tgz" 2>/dev/null; 
-rm -rf "$1_$folder"
-else
-continue
-fi
-done 
-done
-fi
-done
-		
-if [ "${x}" == 0 ] && [ "$1" != "$word" ]; then
-echo "User $1 don't exist"
-x=0 
-exit 2;
-fi
-       
-						
+		if [[ "$2" =~ ([0-9]*minutes|hours|days) ]]
+		then
+			if [[ "$2" =~ [0-9]*hours ]]
+			then
+				minutes=$( echo $2 | grep -o '[0-9]*' )
+				minutes=$(($minutes * 60))
+			elif [[ "$2" =~ [0-9]*days ]]
+			then
+				minutes=$( echo $2 | grep -o '[0-9]*' )
+				minutes=$(($minutes * 1440)) #minutes in day
+			else
+				minutes=$( echo $2 | grep -o '[0-9]*' )
+			fi
+
+			for (( index=1; index <= $countUSERS; index++))
+			do
+				word=$(sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' /etc/passwd | tr '\n' ' ' | cut -d " " -f $index)
+
+				if [ "$1" == "$word" ]
+				then
+					x=1 #user founded
+					while read line
+					do
+						line=$(sed 's/ *//g' <<< "$line")
+
+						IFS=:
+						for i in $line
+						do
+							folder=$(cut -c 2- 2>/dev/null <<< "$i" | tr -d '\n^')
```

Comment:
The `-d` flag of the `tr` command allows for multiple different characters to be deleted, so there's no need to repeatedly execute `tr`.